src/SSEConnection.js

```diff
@@ -0,0 +1,82 @@
+import AbstractConnection from "./AbstractConnection";
+import ConnectionEvent from "./ConnectionEvent";
+import { List } from "immutable";
+
+/**
+ * Server Sent Event Connections
+ */
+export default class SSEConnection extends AbstractConnection {
+  /**
+   * Initializes an Instance of SSEConnection
+   * @constructor
+   * @param {string} url – URL to connect to
+   * @param {object} [options]
+   * @param {boolean} [options.withCredentials=true] – used method
+   */
+  constructor(url, options = {}) {
+    super(url);
+
+    Object.defineProperty(this, "_data", {
+      value: List(),
+      writable: true
+    });
+
+    Object.defineProperty(this, "options", {
+      value: { withCredentials: options.withCredentials || false }
+    });
+  }
+
+  /**
+   * Returns the data received so far
+   * @returns List<String>
+   */
+  get data() {
+    return this._data;
+  }
+
+  /**
+   * Opens the connection to the configured URL
+   */
+  open() {
+    const { withCredentials } = this.options;
+    this.source = new window.EventSource(this.url, { withCredentials });
+
+    this.source.onopen = () => {
+      this.state = SSEConnection.OPEN;
+      this.emit(
+        ConnectionEvent.OPEN,
+        new ConnectionEvent(this, ConnectionEvent.OPEN)
+      );
+    };
+
+    this.source.onmessage = response => {
```

Comment:
`onmessage` behaves differently in browsers. I.e. in Chrome you won't get _all_ messages but only messages without a specific event type. To subscribe to a specific type we need to use `addEvenListener`. We might want to configure the connection with specific events we want to listen to.

Please refer to my PR https://github.com/dcos/dcos-ui/pull/2856/files

Reply:
This means we need to know the types being sent right?

Reply:
That is probably the list in the PR.

Reply:
yeah, see my comments below, they address this issue 👍